mlflow/tracking/client.py

```diff
@@ -30,14 +32,31 @@ def __init__(self, tracking_uri=None, registry_uri=None):
                              defaults to the service set by ``mlflow.tracking.set_tracking_uri``.
         """
         final_tracking_uri = tracking_uri or utils.get_tracking_uri()
-        self.tracking_client = TrackingServiceClient(final_tracking_uri)
-        try:
-            self.registry_client = ModelRegistryClient(registry_uri or final_tracking_uri)
-        except MlflowException as e:
-            if "Unexpected URI" in e.message:
-                self.registry_client = None
-            else:
-                raise e
+        self._tracking_client = TrackingServiceClient(final_tracking_uri)
+        self._registry_uri = registry_uri or final_tracking_uri
+
+    def _get_registry_client(self):
+        """
+        Attempts to create a py:class:`ModelRegistryClient` if one does not already exist.
+
+        :raises: py:class:`mlflow.exceptions.MlflowException` if the py:class:`ModelRegistryClient`
+                 cannot be created. This may occur, for example, when the registry URI refers
+                 to an unsupported store type (e.g., the FileStore).
+        :return: A py:class:`ModelRegistryClient` instance
+        """
+        registry_client_attr = "_registry_client_lazy"
```

Comment:
document it. looks 👻 to me :)

Reply:
Documented! Hopefully the documentation isn't even scarier 👻 :).